library/std/src/panicking.rs

```diff
@@ -571,7 +571,7 @@ fn rust_panic_with_hook(
     // process real quickly as we don't want to try calling it again as it'll
     // probably just panic again.
     if panics > 2 {
-        util::dumb_print(format_args!("thread panicked while processing panic. aborting.\n"));
+        rterr!("thread panicked while processing panic. aborting.\n");
```

Comment:
Is there a particular reason why here in `panicking` `intrinsics::abort` is used instead of `sys::abort_internal`? Otherwise this could also be written with `rtabort!`.

Reply:
`sys::abort_internal` calls `libc::abort()` (at least on unix). `intrinsics::abort` does not rely on libc or the platform, but simply results in executing an invalid instruction. I don't know if there's a reason to avoid libc::abort() here. It'd require some digging through the git history to see if this was changed on purpose, or added before the alternative existed, or something like that. :(

Reply:
I'll look into it

Reply:
I looked through the history, these `intrinsics::abort` were indeed added before `sys::abort_internal` existed. As far as I can tell these could be replaced, along with a safety comment added to `abort_internal` that it's implementation is not allowed to panic (else panicking is stuck in a loop). However there are more occurrences of `intrinsics::abort` in `std` that in my opinion could also be replaced, so I think I'm leaving it like this for now and will address those together in a future PR.